web/pandas/pdeps/0013-logical-type-system.md

```diff
@@ -0,0 +1,196 @@
+
+The existing extension types which will become our "logical types" are:
+
+  - pd.StringDtype()
+  - pd.IntXXDtype()
+  - pd.UIntXXDtype()
+  - pd.FloatXXDtype()
+  - pd.BooleanDtype()
+  - pd.PeriodDtype(freq)
+  - pd.IntervalDtype()
+
+To satisfy all of the types highlighted above, this would require the addition of:
+
+  - pd.DecimalDtype()
+  - pd.DateDtype()
+  - pd.DatetimeDtype(unit, tz)
+  - pd.Duration()
+  - pd.CalendarInterval()
+  - pd.BinaryDtype()
+  - pd.DictDtype()
+  - pd.ListDtype()
+  - pd.StructDtype()
+  - pd.ObjectDtype()
+  - pd.NullDtype()
+
+The storage / backend to each of these types is left as an implementation detail. The fact that ``pd.StringDtype()`` may be backed by Arrow while ``pd.PeriodDtype()`` continues to be a custom solution is of no concern to the end user. Over time this will allow us to adopt more Arrow behind the scenes without breaking the front end for our end users, but _still_ giving us the flexibility to produce data types that Arrow will not implement (e.g. ``pd.ObjectDtype()``).
+
+The methods of each logical type are expected in turn to yield another logical type. This can enable us to smooth over differences between the NumPy and Arrow world, while also leveraging the best of both backends. To illustrate, let's look at some methods where the return type today deviates for end users depending on if they are using NumPy-backed data types or Arrow-backed data types. The equivalent PDEP-13 logical data type is presented as the last column:
+
+| Method             | NumPy-backed result | Arrow Backed result type | PDEP-13 result type            |
+|--------------------|---------------------|--------------------------|--------------------------------|
+| Series.str.len()   | np.float64          | pa.int64()               | pd.Int64Dtype()                |
+| Series.str.split() | object              | pa.list(pa.string())     | pd.ListDtype(pa.StringDtype()) |
+| Series.dt.date     | object              | pa.date32()              | pd.DateDtype()                 |
+
+The ``Series.dt.date`` example is worth an extra look - with a PDEP-13 logical type system in place we would theoretically have the ability to keep our default ``pd.DatetimeDtype()`` backed by our current NumPy-based array but leverage pyarrow for the ``Series.dt.date`` solution, rather than having to implement a DateArray ourselves.
+
+To implement this PDEP, we expect all of the logical types to have at least the following metadata:
+
+    * storage: Either "numpy" or "pyarrow". Describes the library used to create the data buffer
+    * physical_type: Can expose the physical type being used. As an example, StringDtype could return pa.string_view
```

Comment:
i dont think physical_type should be necessary/required for the general case.  AFAICT it is relevant (almost?) exclusively for variants of arrow strings, and could be kept as a keyword for just pd.StringDtype.

Reply:
For the types listed here I think it could also be used by Binary (Arrow binary versus large_binary) and Decimal (decimal128 versus decimal256)

Reply:
While it's true that there are multiple dtypes where this could be relevant, to Brock's point: it will not be relevant for _all_ dtypes. So maybe make this an optional metadata item? (or maybe just give as example how the logical dtype can hide implementation details)